### ISSUE TITLE
Fix User Management

### ISSUE THREAD

**Issue by oskarhinc** [benetech/ServiceNet#212]
User management page is buggy. Deleting users as admin sometimes is trying to delete the wrong user. Lots of ‘Internal Server Error’ and ‘Delete verb not allowed’ are displayed in the console.

**Comment by wojtekbu** [comment]
TODO:

- CRUD should work fine for any user,
- CRUD should work fine for any system account,
- editing roles should work.

**Comment by wojtekbu** [comment]
@oskarhinc the bugs were associated with editing user's system account and removing users in 'Administration → User management' view. Could you test it?

**Comment by johnhbenetech** [comment]
@oskarhinc @wojtekbu 
Not sure if this is already a known issue and the fix isnt commited to master yet, but in the current master branch:
using example passwords:
1. Log in as user, we refer to password as 'passwordold'
2. Go to the change password dialog, enter 'passwordoldl' for old and 'passwordnew' for new
3. Log out
4. Attempt to login, 'passwordnew' does not work and you must use 'passwordold'
5. Once logged in, repeat step 2
6. Password reset will not accept 'passwordold' for old password and instead expects 'passwordnew' (huh!?)
(note) even if you log out before step 6, 'passwordnew' doesnt work for login (despite being expected on password reset screen)

7. If you repeat step 6 and use 'passwordnew' for old and choose a different password the cycle just repeats itself.

So, in summary, it seems like changing a password changes it for the purposes of the 'change password' screen, but the actual password needed to logon always lags to the previous password

**Comment by oskarhinc** [comment]
I've tested it after @wojtekbu's fix, and the problem with caching password and user management is gone. What still can be misleading is I've found out that edit page of an existing user doesn't load its actual system account, so if we would change nothing but press save - the system account will be lost. 

Also, there is some unnecessary request being done after the user is logging out, which results in 401, however, the error is only visible in the console.

**Comment by wojtekbu** [comment]
@oskarhinc user management is fixed.

I didn't manage to remove code 401 from the console without unexpected behavior and since it is not a part of this issue I will skip for now. The 401 in the console does not affect user experience. I have created an issue for that but I think that the priority is to low to deal with that right now.

https://github.com/benetech/ServiceNet/issues/233

**Comment by oskarhinc** [comment]
@johnhbenetech @wojtekbu I've verified it and it well now. All mentioned issues were fixed.
